tests/corpus/test_conversation.py

```diff
@@ -27,22 +28,101 @@ def test_instantiate(self, my_convo, convo_utts, convo_meta):
-            my_convo_read = json.load(f)
-            assert isinstance(my_convo_read, dict)
-            assert my_convo_read == my_convo.asdict()
+            convo_read = json.load(f)
+            assert isinstance(convo_read, dict)
+            assert convo_read == convo.asdict()
+
+
+class TestConversationMetrics:
+    @pytest.mark.parametrize("args, error",
+                             [
+                                 ([0, 0, 1], does_not_raise()),
+                                 ([20, 1, 1], pytest.raises(IndexError)),
+                                 ([0, 50, 50], does_not_raise())
+                             ])
+    def test_subconversation_errors(self, convo, args, error):
+        index, before, after = args
+        with error:
+            convo._subconversation_by_index(index=index,            # noqa W0212
+                                   before=before,
+                                   after=after)
+
+    @pytest.mark.parametrize("args, expected_length",
+                             [
+                                 ([0, 0, 1], 2),
+                                 ([5, 2, 0], 3),
+                                 ([0, 2, 2], 3),
+                                 ([0, 2, None], 3)
+                             ])
+    def test_subconversation_index(self, convo, args, expected_length):
+        index, before, after = args
+        sub = convo._subconversation_by_index(index=index,           # noqa W0212
+                                     before=before,
+                                     after=after)
+        assert isinstance(sub, Conversation)
+        assert len(sub.utterances) == expected_length
+
+    @pytest.mark.parametrize("args, expected_length",
+                             [
+                                 ([0, 0, 0], 2),  # A, B
+                                 ([5, 3000, 3000], 6),  # B,C,E,U,F,H
+                                 ([5, 0, 0], 3),  # C, U, F
+                             ])
+    def test_subconversation(self, convo, args, expected_length):
+        index, before, after = args
+        sub = convo._subconversation_by_time(index=index,           # noqa W0212
+                                     before=before,
+                                     after=after)
+        assert isinstance(sub, Conversation)
+        assert len(sub.utterances) == expected_length
+
+    def test_overlap(self):
+        # entire utterance in window
+        assert Conversation.overlap(80, 120, [90, 110])
+        # beginning of utterance in window
+        assert Conversation.overlap(80, 100, [90, 110])
+        # end of utterance in window
+        assert Conversation.overlap(100, 120, [90, 110])
+        # utterance covers window entirely
+        assert Conversation.overlap(95, 105, [90, 110])
+        assert not Conversation.overlap(
+            120, 140, [90, 110])  # utterance before window
+        assert not Conversation.overlap(
+            70, 80, [90, 110])  # utterance after window
```

Comment:
A minor question, but why not use `@pytest.mark.parameterize` to generate test cases as the previous tests?

Reply:
I've moved the overlap functionality to the Utterances & done as you suggested!